### ISSUE TITLE
[RC2 BLOCKING][CLI MIGRATION BLOCKING] Implicit Imports Break DeepClone

### ISSUE THREAD

**Issue by piotrpMSFT** [Microsoft/msbuild#1431]
[deepclone.zip](https://github.com/Microsoft/msbuild/files/629422/deepclone.zip)

See attached repro. `dotnet restore` and `dotnet run` this one.

Not having looked at the code, I suspect that DeepClone is trying to insert xml relative to the implicit [and therefore not expressed in xml] imports element. Since the element doesn't actually exist, System.Xml throws:

```
Unhandled Exception: System.ArgumentException: The reference node is not a child of this node.
   at System.Xml.XmlNode.InsertAfter(XmlNode newChild, XmlNode refChild)
   at Microsoft.Build.Construction.ProjectElementContainer.AddToXml(ProjectElement child)
   at Microsoft.Build.Construction.ProjectElementContainer.InsertAfterChild(ProjectElement child, ProjectElement reference)
   at Microsoft.Build.Construction.ProjectElementContainer.DeepClone(ProjectRootElement factory, ProjectElementContainer parent)
   at Microsoft.Build.Construction.ProjectElementContainer.DeepClone(ProjectRootElement factory, ProjectElementContainer parent)
   at Microsoft.Build.Construction.ProjectRootElement.DeepClone()
   at Program.Main(String[] args)
```

/cc @rainersigwald @cdmihai @AndyGerlicher @mattgertz

**Comment by piotrpMSFT** [comment]
This blocks CLI and therefore RC2 because CLI deep clones the `dotnet new` template csproj as a skeleton for migrated apps.

**Comment by piotrpMSFT** [comment]
Fix likely is here: https://github.com/Microsoft/msbuild/blob/45cecd3a65090cafc6a09c2dd59ed85d1b805c88/src/XMakeBuildEngine/Construction/ProjectElementContainer.cs#L488

```
Predicate<ProjectElement> siblingIsSameAsChild = _ => _.ExpressedAsAttribute == false && _.IsImplicit == false;
```

**Comment by rainersigwald** [comment]
That does look like the right fix. Testing it out now.

For context, migration always happens through the CLI, right? If so, I'd like to propose taking this fix only for CLI--the risk of being one change apart seems lower than the risk of taking a change to VS this late.

**Comment by rainersigwald** [comment]
Fix is available in packages versioned 15.1.0-preview-000454-01.

@piotrpMSFT

**Comment by piotrpMSFT** [comment]
Fix implemented in #1432 confirmed. Closing this issue.
